schedulers/terraform/argo-workflow/addons.tf

```diff
@@ -113,7 +113,15 @@ module "eks_blueprints_addons" {
   #---------------------------------------
   enable_karpenter                  = true
   karpenter_enable_spot_termination = true
+  karpenter_node = {
+    iam_role_use_name_prefix = false
+    iam_role_name            = "${local.name}-karpenter-node"
+    iam_role_additional_policies = {
+      AmazonSSMManagedInstanceCore = "arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore"
+    }
+  }
```

Comment:
Check this code where we are not defining `iam_role_name`. 

https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L82C1-L96C4

iam role is being passed to the Karpenter provisioner here https://github.com/awslabs/data-on-eks/blob/1e4b51971bed042adf17cb7995cceafb7790a150/analytics/terraform/spark-k8s-operator/addons.tf#L189

Reply:
I saw this, but if the provisioner is in a file separately and not embedded in the addons.tf, then it doesn't know who is "module".